src/zeekconnection.cpp

```diff
@@ -773,11 +781,9 @@ Status ZeekConnection::taskFromZeekEvent(QueryScheduler::Task &task,
   const auto &event_name = event.name();
 
   if (event_name == kHostSubscribeEvent ||
-      event_name == kHostUnsubscribeEvent) {
-
-    return scheduledTaskFromZeekEvent(task, event);
+      event_name == kHostUnsubscribeEvent ||
+      event_name == kHostExecuteEvent) {
 
-  } else if (event_name == kHostExecuteEvent) {
```

Comment:
Is this really a bug? I see some potentially meaningful grouping of related events here (subscribe/unsubscribe; execute). I don't have a strong opinion on this, though.